### PR TITLE
PLAT-115129: Expand framework build scope

### PR DESCRIPTION
**Pull request by JayCanuck**
Targeted against https://github.com/enactjs/dev-utils/pull/59 as that one was already reviewed and didn't want to outright add new content.

Currently the framework build support solely looked in `./node_modules/@enact/*` for source files. Usually ok, but for embedded ui/screenshot tests, that leaves all the local source files of themes left out.

This adds support for local theme source code to be included in framework bundles. This also adds support optionally including polyfills in the framework bundle under the pseduo module `@enact/polyfill`.  This allows ui/screenshot tests to be significantly smaller and build much faster. Also switched from `node-glob` to `fast-glob`, which is faster.

Enact-DCO-1.0-Signed-off-by: Jason Robitaille <jason.robitaille@lge.com>

### CODE REVIEW

File: mixins/framework.js [hunk 1]
```diff
@@ -28,25 +34,59 @@ module.exports = {
 						'**/samples/**/*.*',
 						'**/tests/**/*.*'
```
Comment (webOS101) [inline mixins/framework.js:35]: Is `*.*` needed or is `*` enough?  Do we need to ignore `coverage` here, too?

File: mixins/framework.js [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+		const polyfillFile = helper.polyfillFile(config);
+
 		// Form list of framework entries; Every @enact/* js file as well as react/react-dom
 		config.entry = {
-			enact: glob
-				.sync('@enact/**/*.@(js|jsx|es6)', {
-					cwd: path.resolve(path.join(app, 'node_modules')),
-					nodir: true,
+			enact: fastGlob
+				.sync(['@enact/**/*.@(js|jsx|es6)'], {
```
Comment (webOS101) [inline mixins/framework.js:18]: Seems a string is also accepted here.  Is there a reason to switch to array?  We do it in some places below and not in others.

Comment (JayCanuck) [reply mixins/framework.js:18]: I was under the impression fastglob.sync only took arrays

File: mixins/framework.js (hunk 1 above)
Comment (JayCanuck) [reply mixins/framework.js:35]: This is within the app's node_modules, so I'd hope we weren't publishing coverage files, heh.

File: mixins/framework.js (hunk 2 above)
Comment (webOS101) [reply mixins/framework.js:18]: Good thing it doesn't. :D

**Pull request merged**